timelock/program/src/instruction.rs

```diff
@@ -0,0 +1,590 @@
+use std::{convert::TryInto, mem::size_of};
+
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvar,
+};
+
+use crate::{
+    error::TimelockError,
+    state::{
+        custom_single_signer_timelock_transaction::INSTRUCTION_LIMIT,
+        timelock_config::CONFIG_NAME_LENGTH,
+        timelock_state::{DESC_SIZE, NAME_SIZE},
+    },
+};
+
+/// Used for telling caller what type of format you want back
+#[derive(Clone, PartialEq)]
+pub enum Format {
+    /// JSON format
+    JSON,
+    /// MsgPack format
+    MsgPack,
+}
+impl Default for Format {
+    fn default() -> Self {
+        Format::JSON
+    }
+}
+
+/// Instructions supported by the Timelock program.
+#[derive(Clone)]
+pub enum TimelockInstruction {
+    /// Initializes a new Timelock Program.
+    ///
+    ///   0. `[writable]` Timelock program account pub key.
+    ///   1. `[]` Token program id
+    ///   2. `[]` Rent sysvar
+    InitTimelockProgram,
```

Comment:
Yeah I think I made this back when I was much newer to the ecosystem. I think it might be better to make it per set because then people can bring their own token ecosystem if they so choose.